### PR TITLE
Run dogfood to completion

### PR DESCRIPTION
**Pull request by Jarcho**
Run dogfood on all packages before failing the test. Failing early is painful on lints which trigger on multiple crates.

changelog: None

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ uitest = "test --test compile-test"
 dev = "run --package clippy_dev --bin clippy_dev --manifest-path clippy_dev/Cargo.toml --"
 lintcheck = "run --package lintcheck --bin lintcheck --manifest-path lintcheck/Cargo.toml  -- "
 collect-metadata = "test --test dogfood --features internal -- run_metadata_collection_lint --ignored"
+dogfood = "test --test dogfood"
```
Comment (flip1995) [inline .cargo/config.toml:6]: We have `dev dogfood` with `--fix` support. So I don't think we need another alias.

**Review by flip1995 — APPROVED** [review]
One comment. I'm not opposed with merging it as-is, if you think this is useful. r=me either way.

File: .cargo/config.toml (hunk 1 above)
Comment (Jarcho) [reply .cargo/config.toml:6]: Somehow I missed that. I've been using `test` directly this entire time.

Comment (flip1995) [reply .cargo/config.toml]: I also only noticed it 2 weeks ago or so :smile:

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**